pkg/compactor/block_upload.go

```diff
@@ -504,6 +509,33 @@ func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.U
 		}
 	}
 
+	// check that the blocks doesn't contain down-sampled data
+	if blockMetadata.Thanos.Downsample.Resolution > 0 {
+		return errors.New("block contains downsampled data")
+	}
+	// validate index
+	indexFile := filepath.Join(blockDir, block.IndexFilename)
+	err = block.VerifyIndex(c.logger, indexFile, blockMetadata.MinTime, blockMetadata.MaxTime)
+	if err != nil {
+		return errors.Wrap(err, "error validating block index")
+	}
+	// validate segment files
+	err = filepath.Walk(filepath.Join(blockDir, block.ChunksDirname), func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
```

Comment:
Should this be error? We don't expect any more directories in `chunks` dir.

Reply:
Yeah, we don't expect any more directories, but the first path will be the chunks directory itself. I've removed this entire function for now as part of the PR split, but if I incorporate it again, I'll have it error on any dir that is not chunks.